extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -23,6 +80,20 @@ export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
     this.container_.textContent = this.myText_;
     this.element.appendChild(this.container_);
     applyFillContent(this.container_, /* replacedContent */ true);
+
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win),
+      Services.localizationServiceForOrNull(this.element),
+    ]).then(([storeService, localizationService]) => {
+      this.storeService_ = storeService;
+      this.localizationService_ = localizationService;
+      this.storeService_.subscribe(StateProperty.CURRENT_PAGE_ID, (pageId) => {
+        const pageIndex = this.storeService_.get(
+          StateProperty.CURRENT_PAGE_INDEX
+        );
+        this.updateCta_(pageIndex, pageId);
+      });
```

Comment:
This will be called on every page change. 
We only need this update to happen once.
We only put listeners in layout callback.

How could we approach this to update with SHOPPING_STATE?
A suggestion for how to approach this:
- listen to SHOPPING_STATE
- Check if page includes shopping tag included in the data
- Update CTA text if page has data for the tag

Reply:
A solution involving listening "on the page change" is provided as it is difficult
to listen and update the CTA text after the shopping state AND wait until the CTA fully loads.
Most likely we will be doing other onPageChange ops in the future. Until a better approach can be found, we are going with the onPageChange Listener.

Reply:
I have also tested this a bit further myself and can confirm that as of now, listening for config data changes instead of on the page change will result in buggy (unpredicatble) behavior.

Reply:
There is another way to do this with setting the `data-cta-text` attribute and listening to `SHOPPING_DATA`.
Please give it a try.

Performance is a keystone concept to our codebase. We must do our best to reduce CPU cycles.

Reply:
I see you marked this as resolved but I'm seeing the page change listener in the code.
Did you try the `data-cta-text` approach? I'm happy to brainstorm with you.

Reply:
I have tried the `data-cta-text` while listening to the shopping data and it seems like it is flaky (only works about half of the time) whereas the page listener seems to always work.

I'll update the code with the `data-cta-text` changes  but it seems like the shopping data subscriber is unreliable.